backend/events/tests.py

```diff
@@ -1,3 +1,237 @@
+    # Arrange
+    event_data_copy = event_data.copy()
+    event_data_copy['title'] = 'updated title'
+    event_data_copy['organizer'] = test_user.id
+    event_data_copy['date'] = '2021-01-02'
+
+    # Act
+    response = authenticated_client.put(f'/api/events/{test_event.id}/', event_data_copy, format='json')
+
+    # Assert
+    assert response.status_code == status.HTTP_200_OK
+    assert response.data['title'] == 'updated title'
+    assert response.data['date'] == '2021-01-02'
+
+
+@pytest.mark.django_db
+def testUpdateEventView_validEventId_requestingUserIsNotOwner_shouldThrowForbidden(authenticated_client):
+    # Arrange 
+    account_data_copy = account_data.copy()
+    account_data_copy['email'] = 'changed@email.com'
+    test_user_2 = Account.objects.create(**account_data_copy)
+    event_data_copy = event_data.copy()
+    event_data_copy['organizer'] = test_user_2
+    event = Event.objects.create(**event_data_copy)
+
+    event_data_copy['title'] = 'updated title'
+    event_data_copy['organizer'] = test_user_2.id
+
+    # Act
+    response = authenticated_client.put(f'/api/events/{event.id}/', event_data_copy, format='json')
+
+    # Assert
+    assert response.status_code == status.HTTP_403_FORBIDDEN
+    assert Event.objects.get().title == event_data['title']
+
+
+@pytest.mark.django_db
+def testUpdateEventView_invalidEventId_shouldThrowNotFound(authenticated_client, test_user, test_event):
+    # Arrange
+    event_data_copy = event_data.copy() 
+    event_data_copy['title'] = 'updated title'
+    event_data_copy['organizer'] = test_user.id
+
+    # Act
+    response = authenticated_client.put(f'/api/events/{test_event.id + 1}/', event_data_copy, format='json')
+
+    # Assert
+    assert response.status_code == status.HTTP_404_NOT_FOUND
+    assert Event.objects.get().title == event_data['title']
+
+
+@pytest.mark.django_db
+def testDeleteEventView_validEventId_requestingUserIsOwner_shouldDeleteEvent(test_user, authenticated_client, test_event):
+    # Arrange
+    # Act
+    response = authenticated_client.delete(f'/api/events/{test_event.id}/')
+
+    # Assert
+    assert response.status_code == status.HTTP_204_NO_CONTENT
+    assert Event.objects.count() == 0
+
+
+@pytest.mark.django_db
+def testDeleteEventView_validEventId_requestingUserIsNotOwner_shouldThrowNotAuthorized(authenticated_client):
+    # Arrange   
+    account_data_copy = account_data.copy()
+    account_data_copy['email'] = 'new@email.com'
+    test_user_2 = Account.objects.create(**account_data_copy)
+
+    event_data_copy = event_data.copy()
+    event_data_copy['organizer'] = test_user_2
+    event = Event.objects.create(**event_data_copy)
+
+    # Act
+    response = authenticated_client.delete(f'/api/events/{event.id}/')
+
+    # Assert
+    assert response.status_code == status.HTTP_403_FORBIDDEN
+    assert Event.objects.count() == 1
```

Comment:
better way ig

`assert Event.objects.filter(id=event.id).exists()
`

Reply:
why though?

Reply:
Just a better way to do it I would say